docs/instrumentation/tracing-java-interop.md

```diff
@@ -190,8 +189,43 @@ Java ctx: {traceId=06f5d9112efbe711947ebbded1287a30, spanId=26ed80c398cc039f, ..
 Otel4s ctx: {traceId=06f5d9112efbe711947ebbded1287a30, spanId=26ed80c398cc039f, ...}
 ```
 
-As we can see, the tracing information is in sync now,
-and you can use Java-instrumented libraries within the `useJContext` block. 
+As we can see, the tracing information is now available in ThreadLocal now too.
+Code instrumented using OpenTelemetry's Java API will work inside `blockingWithContext`.
+
+### Calling asynchronous code
+
+When interopting with asynchronous Java APIs:
+
+```scala mdoc:silent
+import scala.concurrent.ExecutionContext
+import cats.effect.Async
+
+// Ensure executing the callback happens on the same thread so Context is correctly propagated and then cleaned up
+def tracedContext(ctx: JContext): ExecutionContext =
+  new ExecutionContext {
+    def execute(runnable: Runnable): Unit = {
+      val scope = ctx.makeCurrent()
+      try {
+        runnable.run()
+      } finally {
+        scope.close()
+      }
+    }
+    def reportFailure(cause: Throwable): Unit =
+      cause.printStackTrace()
+  }
```

Comment:
Sorry, just getting caught up ...

Hmm, I'm concerned by this. If I'm reading correctly you have effectively implemented a `ParasiticExecutionContext`. If you run arbitrary effects on this you may stackoverflow and/or dead-lock.